### PR TITLE
MEPTS-48: TX_CURR changes

### PR DESCRIPTION
**Pull request by edrisse**
Jira Ticket: https://jembiprojects.jira.com/browse/MEPTS-48

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ public SqlCohortDefinition getPatientsWhoHaveNotReturned() {
 		
 		String query = "select patient_id from ( Select p.patient_id,max(encounter_datetime) encounter_datetime from patient p inner join encounter e on e.patient_id=p.patient_id where p.voided=0 and e.voided=0 and e.encounter_type=%s"
 		        + " and e.location_id=:location and e.encounter_datetime<=:onOrBefore group by p.patient_id ) max_frida inner join obs o on o.person_id=max_frida.patient_id where max_frida.encounter_datetime=o.obs_datetime and o.voided=0 and o.concept_id=%s"
-		        + " and o.location_id=:location and datediff(:onOrBefore,o.value_datetime)>=60";
+		        + " and o.location_id=:location and datediff(:onOrBefore,o.value_datetime)>=28";
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java]: Please verify with Pinki that `4weeks=28days`

**Comment by nthfloor** [comment]
One of the point: the excel template for the new columns you added were missing the gender header: `M | F` please add those. Thanks

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java [hunk 2]
```diff
@@ -169,4 +169,14 @@ public CohortDefinition getBaseCohort() {
 		parameters.put("artProgramId", String.valueOf(hivMetadata.getARTProgram().getProgramId()));
 		return generalSql("baseCohort", BaseQueries.getBaseCohortQuery(parameters));
 	}
+
```
[the remark this answers is not in the document]
Comment (edrisse) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java]: Done Nathan

**Comment by edrisse** [comment]
> One of the point: the excel template for the new columns you added were missing the gender header: `M | F` please add those. Thanks

Fixed Nathan

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/BaseQueries.java [hunk 3]
```diff
@@ -24,8 +24,8 @@
 	// they map to concept_id=1369 - TRANSFER FROM OTHER FACILITY
 	// TODO: Query needs to be refactored
 	public static String getBaseCohortQuery(Map<String, String> parameters) {
-		String query = "select p.patient_id from patient p join encounter e on e.patient_id=p.patient_id join person pr on pr.person_id = p.patient_id "
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/BaseQueries.java]: We didn't add this birthdate check here, it was already in the base query. Should we be removing it?

Comment (edrisse) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/BaseQueries.java]: Sorry @nthfloor I'm not sure I understood. We added the `null birthdate` check to base cohort to complete version 1 of the reports and them removed to make the Unknown disaggregation possible.

Comment (nthfloor) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/BaseQueries.java]: The null check was only added at the end on L38 [here](https://github.com/esaude/openmrs-module-eptsreports/blob/36e49169bf5a4732df094f7d3cd6eb5a92c06517/api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/BaseQueries.java#L37-L38). You can see how the base query was joined to person table and the birthdate check was added.

But the birthdate check on L29 was always part of the original query.

Comment (edrisse) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/BaseQueries.java]: Oh, now I understand. Actually I added that line too by mistake. It was my first attempt to implement the exclusion but was not correct. I added it here https://github.com/esaude/openmrs-module-eptsreports/commit/4848f64fd90173d70fbc710b6f254722bd16df9b#diff-4e3e6ce8ab9071a025d7c13568148a14

Comment (nthfloor) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/BaseQueries.java]: Ah I see, ok cool :+1:

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxCurrOldSpec.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	}
+	
+	@Override
+	public String getExcelDesignUuid() {
+		return "8a75c3be-995a-4448-a259-4d57d991b614";
+	}
+	
+	@Override
+	public String getName() {
+		return "TX_CURR Report Old Spec";
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxCurrOldSpec.java]: Please name this `TxCurr60`

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxCurrOldSpec.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+		    ParameterizableUtil.createParameterMappings("endDate=${endDate},location=${location}"));
+		
+		return reportDefinition;
+	}
+	
+	@Override
+	public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+		ReportDesign reportDesign = null;
+		try {
+			reportDesign = createXlsReportDesign(reportDefinition, "TXCURR.xls", "TXCURR_OLD_SPEC.xls_", getExcelDesignUuid(), null);
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxCurrOldSpec.java]: Same here

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxCurrOldSpec.java (hunk 4 above)
Comment (edrisse) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxCurrOldSpec.java]: Done @nthfloor

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxCurrOldSpec.java (hunk 5 above)
Comment (edrisse) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxCurrOldSpec.java]: Done @nthfloor

**Pull request merged**